tests/urlpatterns/tests.py

```diff
@@ -26,23 +26,38 @@ def test_path_lookup_without_parameters(self):
         self.assertEqual(match.url_name, 'articles-2003')
         self.assertEqual(match.args, ())
         self.assertEqual(match.kwargs, {})
+        self.assertEqual(match.route, 'articles/2003/')
 
     def test_path_lookup_with_typed_parameters(self):
         match = resolve('/articles/2015/')
         self.assertEqual(match.url_name, 'articles-year')
         self.assertEqual(match.args, ())
         self.assertEqual(match.kwargs, {'year': 2015})
+        self.assertEqual(match.route, 'articles/<int:year>/')
 
     def test_path_lookup_with_multiple_paramaters(self):
         match = resolve('/articles/2015/04/12/')
         self.assertEqual(match.url_name, 'articles-year-month-day')
         self.assertEqual(match.args, ())
         self.assertEqual(match.kwargs, {'year': 2015, 'month': 4, 'day': 12})
+        self.assertEqual(match.route, 'articles/<int:year>/<int:month>/<int:day>/')
 
     def test_two_variable_at_start_of_path_pattern(self):
         match = resolve('/en/foo/')
         self.assertEqual(match.url_name, 'lang-and-path')
         self.assertEqual(match.kwargs, {'lang': 'en', 'url': 'foo'})
+        self.assertEqual(match.route, '<lang>/<path:url>/')
+
+    def test_re_path(self):
+        match = resolve('/regex/1/')
+        self.assertEqual(match.url_name, 'regex')
+        self.assertEqual(match.kwargs, {'pk': '1'})
+        self.assertEqual(match.route, '^regex/(?P<pk>[0-9]+)/$')
+
+    def test_path_lookup_with_inclusion(self):
+        match = resolve('/included_urls/extra/something/')
+        self.assertEqual(match.url_name, 'inner-extra')
+        self.assertEqual(match.route, r'included_urls/^extra/(?P<extra>\w+)/$')
```

Comment:
I'm wondering whether we want  `r'included_urls/^extra/(?P<extra>\w+)/$` or `r'included_urls/extra/(?P<extra>\w+)/$` (with or without the middle `^`). Any opinion ?

Reply:
Voting without: It's debugging sugar, not a valid route, so we have no way to match incoming requests with a route without stripping it again.
It would be better to have a property `route_components` that would return a sequence of paths/routes that one could join with whatever separator one wants to show the inclusion.

Reply:
Without makes sense to me.

Reply:
I removed it.